worker/src/RTC/IceServer.cpp

```diff
@@ -415,12 +436,17 @@ namespace RTC
 					if (!storedTuple)
 						storedTuple = AddTuple(tuple);
 
-					// Mark it as selected tuple.
-					SetSelectedTuple(storedTuple);
-					// Update state.
-					this->state = IceState::COMPLETED;
-					// Notify the listener.
-					this->listener->OnIceServerCompleted(this);
+					if (nomination > this->remoteNomination)
```

Comment:
Here we are just updating the selected tuple if the STUN packet has a NOMINATION value higher than previous one. However, if the packet doesn't have that attribute and it has USE_CANDIDATE, we are not updating the selected tuple, so this is changing the previous behavior. I'm afraid of the side effects of this specially with Firefox which doesn't implement ICE Lite and I don't know if libwebrtc (Chrome) implements the NOMINATION attribute, does it?

What worries me is that, if the endpoint doesn't support NOMINATION attribute we are behaving differently than before:

- Before this PR any STUN packet with USE_CANDIDATE would update selected tuple.
- After this PR, the selected tuple is not updated if a previous tuple was already selected and a new STUN packet with USE_CANDIDATE is received.

I would like to keep previous behavior in case the STUN packet doesn't have any NOMINATION attribute. What do you think?

Reply:
Yes, we can do this by change `if (nomination > this->remoteNomination)` to `if (nomination >= this->remoteNomination)` in COMPLETE state.

Reply:
Then both cases/conditions (the current one in the PR) and the one you suggest (this one to keep previous behavior) should be needed, right?